src/Analyzer.Utilities/JsonLineNumberResolver.cs

```diff
@@ -0,0 +1,183 @@
+             *       Similar to scenario 1, the line number where the path ends in the original template
+             *       can be taken directly.
+             *     Example:
+             *       expanded: parameters.numberOfCopies.minValue
+             *       original : parameters.numberOfCopies
+             *       used path: parameters.numberOfCopies
+             * 
+             * 3. The path in the expanded template begins with the resources[] array and can't match
+             *    a valid resource in the original template (i.e. the index is too large).
+             *     - There must have been copies of resources added to the expanded template from
+             *       a copy loop in a resource.  Get the name of the copy loop, find
+             *       the resource in the original template defining that name, and replace the resources[] index
+             *       in the expanded template path with the index of the resource that defines the copy.
+             *     Example:
+             *       expanded: resources[9].properties.configuration.encryption
+             *       original : -
+             *       used path: resources[3].properties.configuration.encryption (if resource 9 is a copy of resource 3)
+             * 
+             * 4. The path in the expanded template partially matches a path in the original template,
+             *    but extends beyond a valid path, and the first missing property is a resources[] array
+             *    that is within a higher-level resources[] array.
+             *     - This must be a copied resource from post-processing in the TemplateProcessor library.
+             *       The source resource must be identified in the original template, and the path to that
+             *       resource will replace the path that led to the copied resource.  Then, try matching the
+             *       path again.  (Fortunately, JSON templates do not allow copy loops in child resources.
+             *       https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/copy-resources#iteration-for-a-child-resource)
+             *     Example:
+             *       expanded: resources[5].resources[0].properties.debug
+             *       original : resources[5]
+             *       used path: resources[7].properties.debug (if resource 7 was copied into resource 5 as a child in pre-processing)
+             */
+
+            if (pathInExpandedTemplate == null || originalTemplateRoot == null)
+            {
+                throw new ArgumentNullException(pathInExpandedTemplate == null
+                    ? nameof(pathInExpandedTemplate)
+                    : nameof(originalTemplateRoot));
+            }
+
+            var tokenFromOriginalTemplate = originalTemplateRoot.InsensitiveToken(pathInExpandedTemplate, InsensitivePathNotFoundBehavior.LastValid);
+
+            int? CheckForScenarios1and2()
```

Comment:
How can this be replaces without calling out the scenarios by number

Reply:
Refactored the logic.  It's now more clear what's going on without needing explicit scenario buckets.